linter/src/spawning/spawning.hpp

```diff
@@ -1,15 +1,30 @@
 #pragma once
 
-#include "logging.hpp"
+#include "lint/lint_result.hpp"
 
-#include <unistd.h>
+#include <boost/asio.hpp>
+#include <boost/process.hpp>
 
-#include <string>
+#include <filesystem>
 
 namespace nutc {
 namespace spawning {
 
-void spawn_client(const std::string& uid, std::string& algoid);
+namespace ba = boost::asio;
 
-}
+class LintProcessManager {
+public:
+    const std::filesystem::path& spawner_binary_path();
+
+    LintProcessManager() : io_service_() {}
+
+    ~LintProcessManager() { io_service_.stop(); }
```

Comment:
I may be misunderstanding this. io service/context blocks the thread it runs on - does it run on another thread? If so, why don't we stop that thread? If not, why would we need this here when it's blocking the main thread?